rarible_marketplace_indexer/event/rarible_action.py

```diff
@@ -156,10 +228,171 @@ def _get_match_dto(transaction: RaribleMatchTransaction, datasource: TzktDatasou
         )
 
         return MatchDto(
             internal_order_id=internal_order_id,
             match_amount=AssetValue(transaction.parameter.b_amount),
             match_timestamp=transaction.data.timestamp,
+            taker=transaction.data.sender_address,
+            token_id=None,
+        )
+
+
+class RariblePutBidEvent(AbstractPutBidEvent):
+    platform = PlatformEnum.RARIBLE
+    RariblePutBidTransaction = Transaction[PutBidParameter, RaribleBidsStorage]
+
+    @staticmethod
+    def _get_bid_dto(
+        transaction: RariblePutBidTransaction,
+        datasource: TzktDatasource,
+    ) -> ListDto:
+        internal_order_id = RaribleAware.get_bid_hash(
+            contract=OriginatedAccountAddress(transaction.parameter.pb_asset_contract),
+            bidder=ImplicitAccountAddress(transaction.data.sender_address),
+            token_id=int(transaction.parameter.pb_asset_token_id),
+            asset_class=transaction.parameter.pb_bid_type,
+            asset=transaction.parameter.pb_bid_asset,
+        )
+
+        make = RaribleAware.get_make_dto(
+            sale_type=int(transaction.parameter.pb_bid_type),
+            value=int(transaction.parameter.pb_bid.bid_amount),
+            asset_bytes=bytes.fromhex(transaction.parameter.pb_bid_asset),
+        )
+
+        take = TakeDto(
+            asset_class=AssetClassEnum.MULTI_TOKEN,
+            contract=OriginatedAccountAddress(transaction.parameter.pb_asset_contract),
+            token_id=int(transaction.parameter.pb_asset_token_id),
+            value=AssetValue(transaction.parameter.pb_bid.bid_asset_qty),
         )
+        make_price = AssetValue(make.value / take.value)
+
+        return ListDto(
+            internal_order_id=internal_order_id,
+            maker=ImplicitAccountAddress(transaction.data.sender_address),
+            make_price=make_price,
+            make=make,
+            take=take,
+        )
+
+
+class RariblePutFloorBidEvent(AbstractPutBidEvent):
+    platform = PlatformEnum.RARIBLE
+    RariblePutFloorBidTransaction = Transaction[PutFloorBidParameter, RaribleBidsStorage]
+
+    @staticmethod
+    def _get_bid_dto(
+        transaction: RariblePutFloorBidTransaction,
+        datasource: TzktDatasource,
+    ) -> ListDto:
+        internal_order_id = RaribleAware.get_floor_bid_hash(
+            contract=OriginatedAccountAddress(transaction.parameter.pfb_asset_contract),
+            bidder=ImplicitAccountAddress(transaction.data.sender_address),
+            asset_class=transaction.parameter.pfb_bid_type,
+            asset=transaction.parameter.pfb_bid_asset,
+        )
+
+        make = RaribleAware.get_make_dto(
+            sale_type=int(transaction.parameter.pfb_bid_type),
+            value=int(transaction.parameter.pfb_bid.bid_amount),
+            asset_bytes=bytes.fromhex(transaction.parameter.pfb_bid_asset),
+        )
+
+        take = TakeDto(
+            asset_class=AssetClassEnum.MULTI_TOKEN,
```

Comment:
I think it should AssetClassEnum.COLLECTION here. It's how we distinguish floor bid on the union dide later